### PR TITLE
Add harvester provider support

### PR DESCRIPTION
**Pull request by aaannz**
## What does this PR change?

Inspired by libvirt provider adapted to harvester resources.

~~Until https://github.com/harvester/harvester/issues/2255 is fixed does not work in one go and after initial provisioning terraform needs to reapply to provision salt part.~~ Fixed and released as v0.5.0

### CODE REVIEW

**Comment by aaannz** [comment]
Ready to review, thing to consider is that this duplicates list of images where both libvirt and harvester uses the same cloud-init images (except SLE Micro). So having this list unified should be something to look at.

File: backend_modules/harvester/base/versions.tf [hunk 1]
```diff
@@ -0,0 +1,9 @@
+terraform {
+  required_version = ">= 1.1.5"
```
Comment (rjmateus) [inline backend_modules/harvester/base/versions.tf:2]: Do we really need this terraform version? Currently, sumaform is tested to work with terraform version "1.0.10"

File: backend_modules/harvester/host/versions.tf [hunk 2]
```diff
@@ -0,0 +1,13 @@
+terraform {
+  required_version = ">= 1.1.5"
```
Comment (rjmateus) [inline backend_modules/harvester/host/versions.tf:2]: Same comment about terraform version as before.

File: backend_modules/harvester/base/main.tf [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    opensuse152-ci-pr        = "${var.use_mirror_images ? "http://${var.mirror}" : "https://download.opensuse.org"}/repositories/systemsmanagement:/sumaform:/images:/libvirt/images/opensuse152-ci-pr.x86_64.qcow2"
+    opensuse153-ci-pr        = "http://minima-mirror.mgr.prv.suse.net/repositories/systemsmanagement:/sumaform:/images:/libvirt/images/opensuse153-ci-pr.x86_64.qcow2"
+    opensuse153-ci-pr-client = "${var.use_mirror_images ? "http://${var.mirror}" : "https://download.opensuse.org"}/repositories/systemsmanagement:/sumaform:/images:/libvirt/images/opensuse153-ci-pr-client.x86_64.qcow2"
+    slemicro51-ign  = "${var.use_mirror_images ? "http://${var.mirror}" : "http://download.opensuse.org"}/repositories/systemsmanagement:/sumaform:/images:/microos/images_51/SUSE-MicroOS.x86_64-sumaform.qcow2"
+    slemicro51o-ign = "${var.use_mirror_images ? "http://${var.mirror}" : "http://download.suse.de"}/install/SLE-Micro-5.1-GM/SUSE-MicroOS.x86_64-5.1.0-Default-GM.raw.xz"
+  }
+  network_name       = lookup(var.provider_settings, "network_name", "default")
+  bridge             = lookup(var.provider_settings, "bridge", null)
```
Comment (rjmateus) [inline backend_modules/harvester/base/main.tf:41]: this variable doesn't look to be used.

**Review by rjmateus — COMMENTED** [review]
just some small comments, overall look really good.
thanks for the great contribution

**Comment by rjmateus** [comment]
> Ready to review, thing to consider is that this duplicates list of images where both libvirt and harvester uses the same cloud-init images (except SLE Micro). So having this list unified should be something to look at.

That is something we can think of, on how to share this list.
Maybe a share module which outputs the list of images.

File: backend_modules/harvester/base/main.tf (hunk 3 above)
Comment (aaannz) [reply backend_modules/harvester/base/main.tf:41]: Right! I will remove it.

File: backend_modules/harvester/base/versions.tf (hunk 1 above)
Comment (rjmateus) [reply backend_modules/harvester/base/versions.tf:2]: good for me, and it should work fine in there. 
Also we should update terraform version again :)

**Comment by aaannz** [comment]
I no longer have access to harvester instance and cannot continue on this. Let's close it then.

**Pull request closed without merging**